test/test_reader.py

```diff
@@ -640,3 +640,81 @@ def test_heuristic_for_3D_works_as_expected(dset):
     fh.seek(0)
     reader = create_reader(fh, dimensionality=None)
     assert reader.dimensionality == 3
+
+
+@given(dataset_3d(valid_line_catalog=True))
+@settings(
+    max_examples=10,
+    suppress_health_check=(HealthCheck.too_slow,),
+    deadline=None,
+    phases=(Phase.explicit, Phase.reuse, Phase.generate),
+)
+def test_num_inlines_is_correct(dset):
+    fh = io.BytesIO()
+    write_segy(fh, dset)
+    fh.seek(0)
+    reader = create_reader(fh)
+    assert reader.num_inlines() == dset.num_traces()
+
+
+@given(dataset_3d(valid_line_catalog=True))
+@settings(
+    max_examples=10,
+    suppress_health_check=(HealthCheck.too_slow,),
+    deadline=None,
+    phases=(Phase.explicit, Phase.reuse, Phase.generate),
+)
+def test_num_xlines_is_correct(dset):
+    fh = io.BytesIO()
+    write_segy(fh, dset)
+    fh.seek(0)
+    reader = create_reader(fh)
+    assert reader.num_xlines() == dset.num_traces()
```

Comment:
Is this correct? If dset is a 3D dataset I would expect the number of traces to be equal to the product of the number of inlines and the number of crosslines, assuming the dataset is complete (i.e. all possible inline, crossline combinations have an associated trace). The assertion could be true if the 3d dataset contains only one inline.